libs/s25main/mapGenerator/Resources.cpp

```diff
@@ -0,0 +1,342 @@
+// Copyright (c) 2017 - 2017 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Return To The Roots.
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#include "mapGenerator/Resources.h"
+#include "mapGenerator/Textures.h"
+#include "mapGenerator/DistanceByProperty.h"
+
+namespace rttr {
+namespace mapGenerator {
+
+using namespace libsiedler2;
+
+bool IsHeadQuarterOrHarborPosition(const Map& map, int index)
```

Comment:
What is this doing in Resources? There are already functions like `IsHeadQuarter` somewhere else.

Also use `contains` :)